Discovery/config_dialog.py

```diff
@@ -144,6 +149,14 @@ def write_config(self):
         settings.setValue("scale_expr", self.editScaleExpr.text())
         settings.setValue("bbox_expr", self.editBboxExpr.text())
 
+        if(self.timeCB.isChecked()):
+            settings.setValue("display_time", "0")
+        else:
+            settings.setValue("display_time", self.displayTime.text())
```

Comment:
It would be better to store the display time as an integer in the settings rather than text.

Also, I would suggest to split the configuration into two separate items in settings: 1. is checked (1/0), 2. time. It would be more user friendly (now user would need to check the checkbox and change the value from 0 to something sensible).